File: backend/models/company.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Company 公司模型
type Company struct {
	ID         int    `gorm:"type:integer;primarykey" json:"id"`
	Code       string `gorm:"type:varchar(50);unique;not null" json:"code"`          // 公司代码，用于URL识别
	Name       string `gorm:"type:varchar(100);not null" json:"name"`                // 公司名称
	Logo       string `gorm:"type:varchar(255)" json:"logo"`                         // Logo URL
	ThemeColor string `gorm:"type:varchar(20);default:'#00fff5'" json:"theme_color"` // 主题颜色
	BgColor    string `gorm:"type:varchar(20);default:'#0a0f14'" json:"bg_color"`      // 背景颜色

	// 文案配置
	Title          string `gorm:"type:varchar(100);default:'幸运抽奖'" json:"title"`           // 系统标题
	Subtitle       string `gorm:"type:varchar(200)" json:"subtitle"`                       // 副标题
	WelcomeText    string `gorm:"type:text" json:"welcome_text"`                           // 欢迎语
	RulesText      string `gorm:"type:text" json:"rules_text"`                             // 规则说明
	DrawButtonText string `gorm:"type:varchar(50);default:'点击抽奖'" json:"draw_button_text"` // 抽奖按钮文字
	SuccessText    string `gorm:"type:varchar(200);default:'恭喜中奖！'" json:"success_text"`   // 中奖恭喜语

	// 联系信息
	ContactName  string `gorm:"type:varchar(100)" json:"contact_name"`  // 联系人
	ContactPhone string `gorm:"type:varchar(20)" json:"contact_phone"`  // 联系电话
	ContactEmail string `gorm:"type:varchar(100)" json:"contact_email"` // 联系邮箱

	IsActive  bool      `gorm:"default:true" json:"is_active"` // 是否启用
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// AutoMigrate 自动迁移数据库表（包含Company）
func AutoMigrateWithCompany(db *gorm.DB) error {
	return db.AutoMigrate(
		&Company{},
		&Admin{},
		&User{},
		&PrizeLevel{},
		&Prize{},
		&DrawRecord{},
	)
}

File: frontend/src/utils/company.js
import { ref, computed } from 'vue'
import api from './api'

// 公司配置状态
const companyConfig = ref(null)
const companyCode = ref(null)
const isLoading = ref(false)

// 默认配置
export const defaultConfig = {
  id: 1,
  code: 'DEFAULT',
  name: '默认公司',
  logo: '',
  theme_color: '#00fff5',
  bg_color: '#0a0f14',
  title: '🎉 幸运大抽奖',
  subtitle: 'Luck Lottery System',
  welcome_text: '欢迎参与抽奖活动！',
  rules_text: '每人只能抽一次，祝您好运！',
  draw_button_text: '点击抽奖',
  success_text: '恭喜您获得',
  contact_name: '管理员',
  contact_phone: '400-888-8888',
  contact_email: 'admin@example.com'
}

// 加载公司配置（强制刷新）
export async function loadCompanyConfig(code) {
  if (!code) {
    // 如果没有提供code，尝试从URL获取
    // 注意：参数可能在hash中，如：#/lottery?company=4000
    let urlCode = null

    // 先尝试从hash中获取
    if (window.location.hash.includes('?')) {
      const hashParts = window.location.hash.split('?')
      if (hashParts.length > 1) {
        const hashParams = new URLSearchParams(hashParts[1])
        urlCode = hashParams.get('company')
      }
    }

    // 如果hash中没有，尝试从search中获取
    if (!urlCode && window.location.search) {
      const urlParams = new URLSearchParams(window.location.search)
      urlCode = urlParams.get('company')
    }

    code = urlCode || 'DEFAULT'
  }

  companyCode.value = code
  isLoading.value = true

  try {
    // 每次都重新获取，不使用缓存（使用api实例，自动带token）
    const response = await api.get('/api/company', {
      params: { code: code }
    })
    companyConfig.value = response
    return response
  } catch (error) {
    // 如果加载失败，使用默认配置
    companyConfig.value = defaultConfig
    return defaultConfig
  } finally {
    isLoading.value = false
  }
}

// 获取当前公司配置
export function useCompany() {
  const setCompanyCode = (code) => {
    companyCode.value = code
  }

  return {
    companyConfig: computed(() => companyConfig.value || defaultConfig),
    companyCode: computed(() => companyCode.value || 'DEFAULT'),
    setCompanyCode,
    isLoading: computed(() => isLoading.value),
    themeColor: computed(() => companyConfig.value?.theme_color || defaultConfig.theme_color),
    bgColor: computed(() => companyConfig.value?.bg_color || defaultConfig.bg_color),
    title: computed(() => companyConfig.value?.title || defaultConfig.title),
    subtitle: computed(() => companyConfig.value?.subtitle || defaultConfig.subtitle),
    welcomeText: computed(() => companyConfig.value?.welcome_text || defaultConfig.welcome_text),
    rulesText: computed(() => companyConfig.value?.rules_text || defaultConfig.rules_text),
    drawButtonText: computed(() => companyConfig.value?.draw_button_text || defaultConfig.draw_button_text),
    successText: computed(() => companyConfig.value?.success_text || defaultConfig.successText)
  }
}

// 重置公司配置
export function resetCompanyConfig() {
  companyConfig.value = null
  companyCode.value = null
}
